src/components/report-suspicious-activity/report-suspicious-activity-controller.ts

```diff
@@ -0,0 +1,124 @@
+import { Request, Response, NextFunction } from "express";
+import { PATH_DATA } from "../../app.constants";
+import { formatEvent } from "../../utils/activityHistory";
+import { getAppEnv, getDynamoActivityLogStoreTableName } from "../../config";
+import { DynamoDB } from "aws-sdk";
+import { dynamoDBService } from "../../utils/dynamo";
+import { ActivityLogEntry } from "../../utils/types";
+import { logger } from "../../utils/logger";
+import assert from "node:assert";
+
+const activityLogDynamoDBRequest = (
+  subjectId: string,
+  eventId: string
+): DynamoDB.Types.QueryInput => ({
+  TableName: getDynamoActivityLogStoreTableName(),
+  KeyConditionExpression: "user_id = :user_id AND event_id = :event_id ",
+  ExpressionAttributeValues: {
+    ":user_id": { S: subjectId },
+    ":event_id": { S: eventId },
+  },
+});
+
+export async function reportSuspiciousActivityGet(
+  req: Request,
+  res: Response,
+  next: NextFunction
+): Promise<void> {
+  let alreadyReported;
+  try {
+    assert.ok(
+      req.query.event !== undefined && req.query.event !== "",
+      "mandatory query parameter 'event' missing from request"
+    );
+    assert.ok(
+      req.query.reported !== undefined && req.query.reported !== "",
+      "mandatory query parameter 'reported' missing from request"
+    );
+    try {
+      assert.strictEqual(req.query.reported.toString(), "true");
+      alreadyReported = true;
+    } catch (err) {
+      try {
+        assert.strictEqual(req.query.reported.toString(), "false");
+        alreadyReported = false;
+      } catch (err) {
+        req.log.error(
+          `Query parameter "reported" was <${req.query.reported}> but expected "true" or "false" `
+        );
+        return next();
+      }
+    }
```

Comment:
Why do we need the the `reported` query parameter? Can we not get the value from the DB and render accordingly?

If we do need the query parameter, then this code is a bit hard to read, could something like this work...
      
```suggestion
    assert.match(req.query.reported.toString(), /^(true|false)$/, `Query parameter "reported" was <${req.query.reported}> but expected "true" or "false" `);
    alreadyReported = req.query.reported.toString() === "true"
```